src/main/java/it/gov/pagopa/payhub/activities/connector/classification/AssessmentClassificationServiceImpl.java

```diff
@@ -0,0 +1,138 @@
+				iud
+		);
+		if (collectionModelAssessmentsDetail == null ||
+				collectionModelAssessmentsDetail.getEmbedded() == null ||
+				collectionModelAssessmentsDetail.getEmbedded().getAssessmentsDetails() == null ||
+				collectionModelAssessmentsDetail.getEmbedded().getAssessmentsDetails().isEmpty()
+		) {
+			log.info("Assessment details not found for organizationId: {}, iuv: {}, iud: {}", organizationId, iuv, iud);
+			return null;
+		}
+		List<AssessmentsDetail> assessmentsDetailList = collectionModelAssessmentsDetail.getEmbedded().getAssessmentsDetails();
+
+		Assessments assessment = assessmentService.findAssessment(assessmentsDetailList.getFirst().getAssessmentId());
+		if (assessment == null) {
+			log.info("Assessment not found for organizationId: {}, iuv: {}, iud: {}", organizationId, iuv, iud);
+			return null;
+		}
+
+		CollectionModelClassification collectionModelClassification = classificationService.findAllByOrganizationIdAndIuvAndIud(
+				organizationId,
+				iuv,
+				iud
+		);
+		if (collectionModelClassification == null ||
+				collectionModelClassification.getEmbedded() == null ||
+				collectionModelClassification.getEmbedded().getClassifications() == null ||
+				collectionModelClassification.getEmbedded().getClassifications().isEmpty()) {
+			log.info("Classifications not found for organizationId: {}, iuv: {}, iud: {}", organizationId, iuv, iud);
+			return null;
+		}
+		List<Classification> classificationList = collectionModelClassification.getEmbedded().getClassifications();
+
+		List<AssessmentsDetail> updatedAssessmentsDetailList = classifyAssessmentsDetails(assessmentsDetailList, classificationList);
+
+		return assessmentMapper.map(assessment, updatedAssessmentsDetailList);
+	}
+
+	private List<AssessmentsDetail> classifyAssessmentsDetails(List<AssessmentsDetail> assessmentsDetailList, List<Classification> classificationList) {
+		ClassificationLabel classificationLabel = null;
+		LocalDate classificationLabelDate = null;
+		for (Classification classification: classificationList) {
+			switch (classification.getLabel()) {
+				case RT_NO_IUF, RT_NO_IUD -> {
+					if(classificationLabel == null || ClassificationLabel.PAID.compareTo(classificationLabel) >= 0) {
+						classificationLabel = ClassificationLabel.PAID;
+						classificationLabelDate = classification.getPayDate();
+					}
+				}
+				case RT_IUF, IUF_TES_DIV_IMP -> {
+					classificationLabel = ClassificationLabel.REPORTED;
+					classificationLabelDate = classification.getRegulationDate();
+				}
+				case RT_TES, RT_IUF_TES, IUD_RT_IUF_TES -> {
+					return updateAssessmentsDetails(assessmentsDetailList, ClassificationLabel.CASHED, classification.getBillDate());
+				}
+				default ->
+					log.debug("Unused label for assessment classification: label {}, classificationId {}", classification.getLabel(), classification.getClassificationId());
+			}
+		}
+		return updateAssessmentsDetails(assessmentsDetailList, classificationLabel, classificationLabelDate);
```

Comment:
let's call the save logic on the main method, the moved logic should just return the label to assign.
the dates to set on the assessment can be read by any classification record